catalyst-gateway/bin/src/db/event/mod.rs

```diff
@@ -0,0 +1,237 @@
+//! Catalyst Election Database crate
+use std::{
+    str::FromStr,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, OnceLock,
+    },
+};
+
+use bb8::Pool;
+use bb8_postgres::PostgresConnectionManager;
+use tokio_postgres::{types::ToSql, NoTls, Row};
+use tracing::{debug, debug_span, error, Instrument};
+
+use crate::settings::Settings;
+
+pub(crate) mod error;
+pub(crate) mod legacy;
+pub(crate) mod schema_check;
+
+/// Database version this crate matches.
+/// Must equal the last Migrations Version Number.
+pub(crate) const DATABASE_SCHEMA_VERSION: i32 = 9;
+
+/// Postgres Connection Manager DB Pool
+type SqlDbPool = Arc<Pool<PostgresConnectionManager<NoTls>>>;
+
+/// Postgres Connection Manager DB Pool Instance
+static EVENT_DB_POOL: OnceLock<SqlDbPool> = OnceLock::new();
+
+/// Is Deep Query Analysis enabled or not?
+static DEEP_QUERY_INSPECT: AtomicBool = AtomicBool::new(false);
+
+/// The Catalyst Event SQL Database
+pub(crate) struct EventDB {}
```

Comment:
Do we need struct ? Maybe we can remove it ?

Reply:
Yes, it has many implemented methods.
The fields internally were removed when the connection handling was refactored, eliminating the struct itself would create an even bigger PR, and it may end up with fields again in the future.  I would rather not consider removing it until the code is complete and all the obsolete code has been removed.  There is enough to do without the refactor that would knock on throughout the code.